### PR TITLE
Add unknown to all enums

### PR DESCRIPTION
**Pull request by abrooksv**
* Make from method return unknown if none match instead of throwing error
* Remove un-needed toString when type is already a string
* Add another suppression to make it ignore "could be private"

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/jetbrains/src/test/resources/testResultOutput [hunk 1]
```diff
@@ -32,12 +31,14 @@ enum class Result(
 
     DashSeparated("dash-separated"),
 
-    UnderscoreSeparated("underscore_separated");
+    UnderscoreSeparated("underscore_separated"),
+
+    Unknown("unknown");
```
Comment (kiiadi) [inline telemetry/jetbrains/src/test/resources/testResultOutput:36]: Assume we should follow this pattern with all languages?

Comment (abrooksv) [reply telemetry/jetbrains/src/test/resources/testResultOutput:36]: Probs? I havent looked at the others yet.

In JetBrains we failed when we added new lambda runtimes and forgot telemetry so I ant to fall back to unknown in that case

Comment (abrooksv) [reply telemetry/jetbrains/src/test/resources/testResultOutput:36]: TS seems to be `foo|bar` so it will fail to compile so dont think we need it there

Comment (abrooksv) [reply telemetry/jetbrains/src/test/resources/testResultOutput:36]: C# does not generate a static from method, so probably not needed there either?

Comment (kiiadi) [reply telemetry/jetbrains/src/test/resources/testResultOutput:36]: what was the failure in JetBrains? was it a build time thing? cos if so - maybe it should fail so we see it....

File: telemetry/jetbrains/src/test/resources/testResultOutput [hunk 2]
```diff
@@ -32,12 +31,14 @@ enum class Result(
...
 
     override fun toString(): String = value
 
     companion object {
-        fun from(type: Any): Result = values().first { it.value == type.toString() }
```
Comment (hunterwerlla) [inline telemetry/jetbrains/src/test/resources/testResultOutput:40]: Why does this no longer take any `Any`?

File: telemetry/jetbrains/src/test/resources/testResultOutput (hunk 1 above)
Comment (abrooksv) [reply telemetry/jetbrains/src/test/resources/testResultOutput:36]: Nah, we converted the Lambda Runtime enum value to this version though the value `([Telemetry]Runtime.from([LambdaRuntime]runtime.toString)`

File: telemetry/jetbrains/src/test/resources/testResultOutput (hunk 2 above)
Comment (abrooksv) [reply telemetry/jetbrains/src/test/resources/testResultOutput:40]: Why would it take an Any? 
We forced the value to be a String, so when would we not want it to take a string in the from? 

Implicitly relying on a toString to match the value of this seems more dirty than making the call site be aware that it owns the conversion

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**